### ISSUE TITLE
C-API Array function RARRAY_PTR allows memcpying between arrays fails on macOS native

### ISSUE THREAD

**Issue by chrisseaton** [oracle/truffleruby#1828]
This fails reliably in native on macOS, including locally.

```
1)
--
  | C-API Array function RARRAY_PTR allows memcpying between arrays FAILED
  | Expected [1, 2, 1] == [1, 2, 3]
  | to be truthy but was false
  | /Users/anka/app/spec/ruby/optional/capi/array_spec.rb:257:in `block (3 levels) in <top (required)>'
  | /Users/anka/app/spec/ruby/optional/capi/array_spec.rb:15:in `<top (required)>'
```

**Comment by chrisseaton** [comment]
Actually it's just flaky but it fails most of the time.

**Comment by eregon** [comment]
I cannot reproduce on Linux.

**Comment by aardvark179** [comment]
I think I've found the cause, and have managed to run those tests 50 times locally without failure.

**Comment by eregon** [comment]
@aardvark179 fixed it in 4bf3272849b553fc39092dc3103b6cf32aa8088d.
